utils/voting.py

```diff
@@ -35,13 +36,20 @@ def create_vote(
     verbose: bool = False,
     cast_vote: bool = False,
     executes_if_decided: bool = False,
+    description: str = "",
 ) -> Tuple[int, Optional[TransactionReceipt]]:
     vote_desc_str = ""
     for v in vote_items.keys():
         vote_desc_str += f"{v};\n "
     if len(vote_desc_str) > 0:
         vote_desc_str = f"Omnibus vote: {vote_desc_str[:-3]}."
 
+    cid = ""
+    if description:
+        cid = upload_str_to_ipfs(description.strip())
+
+    vote_desc_str = f"{vote_desc_str}\n{cid}"
```

Comment:
What would happen for an empty description?
What would happen on re-run?
What would happen on fork-test-run (would the thing be uploaded every single time)?

Reply:
- What would happen for an empty description?

no description - nothing to upload CID will be empty

- What would happen on re-run?
- What would happen on fork-test-run (would the thing be uploaded every single time)?
it will try to upload every time but same description will not create new urls